pkg/data/db/open.go

```diff
@@ -0,0 +1,32 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/cenkalti/backoff"
+
+	// import postgres driver
+	_ "github.com/lib/pq"
+)
+
+// Open opens a postgres database and retries until ctx.Done()
+func Open(ctx context.Context, connStr string) (db *sql.DB, err error) {
```

Comment:
this could be misleading because it is postgres only .   a lot of the db types are also most likely postgres only. Perhaps there should be a pq package here

Reply:
I made it universal.